solvebio/query.py

```diff
@@ -435,9 +435,8 @@ def __len__(self):
                  SELECT * FROM <table> [WHERE condition] [LIMIT number]
               )
         """
-        if getattr(self, '_total', None) == float('inf'):
-            # TODO: Is there a better way?
-            return 999999999
+        if getattr(self, '_is_join', False):
+            return len(self._buffer)
```

Comment:
So the idea here is that the buffer is never empty if its a join, right?

Reply:
It does not matter, if query_a is empty the result is empty join. If query_b is empty the result is all rows from query_a but with empty values.